desktop/core/src/desktop/lib/importer/api.py

```diff
@@ -0,0 +1,281 @@
+
+
+@api_view(["POST"])
+@parser_classes([MultiPartParser])
+@api_error_handler
+def local_file_upload(request: Request) -> Response:
+  """Handle the local file upload operation.
+
+  This endpoint allows users to upload a file from their local system.
+  Uploaded file is validated using LocalFileUploadSerializer and processed using local_file_upload operation.
+
+  Args:
+    request: Request object containing the file to upload
+
+  Returns:
+    Response containing the result of the local upload operation, including:
+      - file_path: Path where the file was saved (if successful)
+
+  Note:
+    - File size limits apply based on MAX_LOCAL_FILE_SIZE_UPLOAD_LIMIT configuration.
+    - File type restrictions apply based on RESTRICT_LOCAL_FILE_EXTENSIONS configuration.
+  """
+
+  serializer = LocalFileUploadSerializer(data=request.data)
+
+  if not serializer.is_valid():
+    return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+  uploaded_file = serializer.validated_data["file"]
+
+  LOG.info(f"User {request.user.username} is uploading a local file: {uploaded_file.name}")
+  result = operations.local_file_upload(uploaded_file, request.user.username)
+
+  return Response(result, status=status.HTTP_201_CREATED)
+
+
+@api_view(["GET"])
+@parser_classes([JSONParser])
+@api_error_handler
+def guess_file_metadata(request: Request) -> Response:
+  """Guess the metadata of a file based on its content or extension.
+
+  This API endpoint detects file type and extracts metadata properties such as
+  delimiters for CSV files or sheet names for Excel files.
+
+  Args:
+    request: Request object containing query parameters:
+      - file_path: Path to the file
+      - import_type: 'local' or 'remote'
+
+  Returns:
+    Response containing file metadata including:
+      - type: File type (e.g., excel, csv, tsv)
+      - sheet_names: List of sheet names (for Excel files)
+      - field_separator: Field separator character (for delimited files)
+      - quote_char: Quote character (for delimited files)
+      - record_separator: Record separator (for delimited files)
+  """
+  serializer = GuessFileMetadataSerializer(data=request.query_params)
+
+  if not serializer.is_valid():
+    return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+  validated_data = serializer.validated_data
+  file_path = validated_data["file_path"]
+  import_type = validated_data["import_type"]
+
+  try:
+    metadata = operations.guess_file_metadata(
+      file_path=file_path, import_type=import_type, fs=request.fs if import_type == "remote" else None
+    )
+
+    return Response(metadata, status=status.HTTP_200_OK)
+
+  except ValueError as e:
+    return Response({"error": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+  except Exception as e:
+    # Handle other errors
+    LOG.exception(f"Error guessing file metadata: {e}", exc_info=True)
+    return Response({"error": str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
Will this expose the stack trace to browser?

Reply:
No